spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/FailoverClientConnectionFactoryTests.java

```diff
@@ -193,10 +201,52 @@ public void testFailoverAllDead() throws Exception {
 		FailoverClientConnectionFactory failoverFactory = new FailoverClientConnectionFactory(factories);
 		failoverFactory.start();
 		GenericMessage<String> message = new GenericMessage<String>("foo");
-		failoverFactory.getConnection().send(message);
+		assertThatExceptionOfType(UncheckedIOException.class).isThrownBy(() ->
+			failoverFactory.getConnection().send(message));
 		Mockito.verify(conn2).send(message);
 	}
 
+	@LogLevels(level = "DEBUG", classes =
+			{ FailoverClientConnectionFactory.class, TcpNetClientConnectionFactory.class, TcpNetConnection.class },
+			categories =
+			"org.springframework.integration.ip.tcp.connection.FailoverClientConnectionFactory$FailoverTcpConnection")
```

Comment:
Do we still need a diagnostics then?

Reply:
Still failing - will remove when resolved.